pkg/controller/templateupdaterequest/templateupdaterequest_controller.go

```diff
@@ -0,0 +1,320 @@
+	if found && completeCond.Status == corev1.ConditionTrue {
+		deleted, requeueAfter, err := r.checkTransitionTimeAndDelete(logger, tur, completeCond)
+		if deleted {
+			return reconcile.Result{}, nil
+		}
+		if err != nil {
+			return reconcile.Result{}, r.updateStatusConditions(logger, tur, toFailure(errs.Wrap(err, "unable to delete the resource")))
+		}
+		return reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: requeueAfter,
+		}, nil
+	}
+
+	// lookup the MasterUserRecord with the same name as the TemplateUpdateRequest tur
+	mur := &toolchainv1alpha1.MasterUserRecord{}
+	err = r.client.Get(context.TODO(), request.NamespacedName, mur)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// MUR object not found, could have been deleted after reconcile request.
+			// Marking this TemplateUpdateRequest as failed
+			return reconcile.Result{}, r.updateStatusConditions(logger, tur, toFailure(err))
+		}
+		// Error reading the object - requeue the request.
+		logger.Error(err, "Unable to get the MasterUserRecord associated with the TemplateUpdateRequest")
+		return reconcile.Result{}, err
+	}
+	if len(tur.Status.SyncIndexes) == 0 {
+		// need to be "captured" before updating the MURs
+		syncIndexes := syncIndexes(tur.Spec.TierName, *mur)
+		// if the TemplateUpdateRequest was just created (ie, `Status.SyncIndexes` is empty), then we should update the associated MasterUserRecord
+		// and retain its current syncIndexex in the status
+		if err = r.updateTemplateRefs(logger, *tur, mur); err != nil {
+			logger.Error(err, "Unable to update the MasterUserRecord associated with the TemplateUpdateRequest")
+			return reconcile.Result{}, r.updateStatusConditions(logger, tur, toFailure(err))
+		}
+		// update the TemplateUpdateRequest status and requeue to keep tracking the MUR changes
+		logger.Info("MasterUserRecord update started. Updating TemplateUpdateRequest status accordingly")
+		if err = r.setStatusConditionsUpdating(logger, tur, syncIndexes); err != nil {
+			logger.Error(err, "Unable to update the TemplateUpdateRequest status")
+			return reconcile.Result{}, err
+		}
+		// now requeue to track progress
+		return reconcile.Result{Requeue: true, RequeueAfter: ProgressDelay}, nil
+	}
+	// otherwise, we should compare the sync indexes of the MasterUserRecord until all tier-related values changed
+	if r.compareSyncIndexes(logger, *tur, *mur) {
+		// once MasterUserRecord is up-to-date, we can delete this TemplateUpdateRequest
+		logger.Info("MasterUserRecord is up-to-date. Marking the TemplateUpdateRequest as complete")
+		// if err := r.client.Status Delete(context.TODO(), tur); err != nil {
+		// 	logger.Error(err, "Unable to delete the TemplateUpdateRequest")
+		// 	return reconcile.Result{}, err
+		// }
+		return reconcile.Result{Requeue: true, RequeueAfter: DeletionTimeout}, r.updateStatusConditions(logger, tur, toBeComplete())
```

Comment:
you don't check if the MUR.status is `Provisioned` - we should wait until it is in provisioned state, shouldn't we?

Reply:
done in 90f3732. Or at least, I check that the MUR is in a `Ready/true` condition.